draft-ietf-quic-tls.md

```diff
@@ -1575,6 +1018,47 @@ version of QUIC defined in {{QUIC-TRANSPORT}} is used.
 The quic_transport_parameters extension is carried in the ClientHello and the
 EncryptedExtensions messages during the handshake.
 
+While the transport parameters are technically available prior to the
+completion of the handshake, they cannot be fully trusted until the handshake
+completes, and reliance on them should be minimized.
+However, any tampering with the parameters will be detected
+when the handshake completes.
+
+
+## QUIC Max Crypto Data Extension {#max_crypto_data}
+
+When QUIC provides TLS messages via the CRYPTO_HS frame, one TLS message may be
+fragmented across different packets.  Thus TLS might have to buffer data.  TLS
```

Comment:
A QUIC implementation needs to pass the input to TLS stack every time it receives a CRYPTO_HS frame, because it cannot tell where the end of a TLS message is.

I can understand the fact that you want to minimize the calls, but even in such case, I think per-packet interaction with the TLS stack just for the handshake messages would not be an issue. Note that heavy operations including public key crypto are involved in the handshake.

Reply:
Still, I think that sentence could be removed without loss of correctness.  The message might (not may) be fragmented.  Implementations MAY limit the amount of out-of-order data they will tolerate.

Reply:
Removed.